pkg/quota/image/registry.go

```diff
@@ -11,19 +11,29 @@
 package image
 
 import (
+	"fmt"
+	"time"
+
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/cache"
 	"k8s.io/kubernetes/pkg/quota"
 	"k8s.io/kubernetes/pkg/quota/generic"
 
 	osclient "github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const (
+	imageTTL time.Duration = time.Minute * 30
```

Comment:
@smarterclayton Images don't change that much if at all. Build often takes several minutes to complete. And resource quota sync takes 5 minutes by default. If we want to take advantage of previous admission checks or quota sync, we should make it at least 6 minutes. I'd round it to 10.

What I'm worried more about is that there are 3 separate image caches:

- in `OriginResourceQuota` admission plugin
- in `ImageLimitRange` admission plugin
- in resource quota controller

Which triples the memory usage and decreases the change for a hit. Shall we perhaps add an `ImageCache` to a `MasterConfig` and share it between them?